0036-spsp-pull-payments/0036-spsp-pull-payments.md

```diff
@@ -0,0 +1,135 @@
+
+## Introduction
+
+### Motivation
+
+The Simple Payment Setup Protocol (SPSP) only describes how the details required for a STREAM connection are exchanged and how a simple push payment is made. However, it lacks a detailed explaination of how a pull payment is conducted using this connection. 
+
+### Scope
+
+This document specifies how to conduct a pull payment once a STREAM connection has been set up. It is intended for use by end-user applications.
+
+### Operation Overview
+
+Any SPSP server will expose an HTTPS endpoint called the SPSP Endpoint. The client can query this endpoint to get information about about the server's connection details, namely ILP address and a shared secret. The endpoint MAY also contain additional information for displaying purposes. The client can set up a STREAM connection and pull multiple ILP payments using the details provided by the server. 
+
+## Model of Operation
+
+### Creating a token
+
+A pull payment token SHOULD be be opaque. For the purpose of offline generation, it MAY include negotiated pull payment agreement parameters that SHOULD become obsolete once the SPSP endpoint corresponding to that token has been created. The SPSP server SHOULD store the negotiated pull payment agreement parameters.
+
+The pull payment agreement parameters that SHOULD be negotiated are those defined in the `agreement` object within the [Response Body](#Response-Body). 
+
+### Conducting the pull payment
+
+We assume that the client knows the server's SPSP endpoint (see [Payment Pointers](../0026-payment-pointers/0026-payment-pointers.md)).
+
+1. The user's SPSP client queries the server's SPSP Endpoint.
+
+2. The SPSP endpoint responds with the server info, including the server's ILP address (`destination_account`) and the shared secret (`shared_secret`) to be used in STREAM. 
+    * The `destination_account` SHOULD be used as the STREAM destinationAccount.
+    * The `shared_secret` SHOULD be decoded from base64 and used as the STREAM sharedSecret.
+
+3. The SPSP client establishes a STREAM connection to the server using the server's ILP address and shared secret and constructs a stream to the client on the ILP/STREAM connection.
+
+4. The SPSP server begins sending ILP packets to fulfill the payment.
+    1. The server will adjust their `sendMax` to reflect the amount they're willing to send.
+        * `sendMax` SHOULD be derived from the pull payment agreement parameters.
+    2. The client will adjust their `receiveMax` to reflect the amount they're willing to receive.
+        * `receiveMax` MAY be `Infinity`.
+    3. The client's and server's STREAM modules will move as much value as possible while staying inside these bounds.
+    4. If the server reaches their `sendMax`, they end the stream and the connection. If the client reaches their `receiveMax`, they will end the stream and the connection.
```

Comment:
Is the client allowed to re-open a connection using the same token? Or is this one-time use? If both versions are allowed, how can you tell the difference?

Reply:
I was thinking of the token kind of like an id for the pull agreement with an expiry time. As long as you have it and it has not expired, you can use it to open a new connection and pull. If the endpoint is in the money, you will actually pull value, otherwise you pull 0. 
But maybe that is too insecure?